src/pyrokinetics/diagnostics/__init__.py

```diff
@@ -391,6 +391,227 @@ def ideal_ballooning_solver(self, theta0: float = 0.0):
 
         return gamma
 
+    def compute_half_disp(
+        self,
+        time: float,
+    ):
+        """
+        Returns the radial displacement of a magnetic field line
+        after half poloidal turn, which is used to investigate the
+        occurance of a nonzonal transition. See Pueschel et al.
+        Phys. Rev. Lett. 110, 155005 (2013) for details.
+        This routine may take a while.
+
+        Available for CGYRO, GENE and GS2 nonlinear simulations
+
+        You need to load the output files of a simulation
+        berore calling this function.
+
+        Parameters
+        ----------
+        time: float, time reference
+
+        Returns
+        -------
+        displacement: numpy.ndarray, 2D array of shape (nx, ny) containing the
+               displacement of each magnetic filed line starting at (x, y).
+
+        Raises
+        ------
+        NotImplementedError: if `Pyro.gk_code` is not ``CGYRO``, ``GENE`` or ``GS2``
+        RuntimeError: in case of linear simulation
+        """
+
+        import pint_xarray  # noqa
+
+        if self.pyro.gk_output is None:
+            raise RuntimeError(
+                "Diagnostics: Please load gk output files (Pyro.load_gk_output)"
+                " before using any diagnostic"
+            )
+
+        if self.pyro.gk_code not in ["CGYRO", "GS2", "GENE"]:
```

Comment:
Why is it only available for these codes? The 4D fields are spit out for other codes too (STELLA/GKW)

Reply:
Yes, that's legacy. I will took that out.